src/models/barotropic.jl

```diff
@@ -0,0 +1,79 @@
+export BarotropicModel, initialize!
+
+"""
+$(SIGNATURES)
+The BarotropicModel struct holds all other structs that contain precalculated constants,
+whether scalars or arrays that do not change throughout model integration.
+$(TYPEDFIELDS)"""
+Base.@kwdef mutable struct BarotropicModel{
+    NF<:AbstractFloat,
+    PL<:AbstractPlanet,
+    AT<:AbstractAtmosphere,
+    CO<:AbstractCoriolis,
+    FR<:AbstractForcing,
+    DR<:AbstractDrag,
+    IC<:InitialConditions,
+    TS<:AbstractTimeStepper,
+    ST<:SpectralTransform{NF},
+    HD<:AbstractHorizontalDiffusion,
+    GE<:AbstractGeometry,
+    DS<:AbstractDevice,
```

Comment:
I'd say it's best to be consistent and always have the `AbstractDevice` as the second parameter. Will make it easier to write rountines that dispatch on it when we tackle the GPU models

Reply:
Yes, that's good call. But also something we can change anytime: At the moment we do not dispatch over the parameters of model types, so changing their order will not cause any issues. The change here is first intended to remove abstract types from models so that the compiler can directly reason about all components of the model